crates/sui-framework/sources/coin.move

```diff
@@ -125,7 +125,8 @@ module sui::coin {
         let i = 0;
         let len = vector::length(&coins);
         while (i < len) {
-            let coin = vector::remove(&mut coins, i);
+            // remove first from front
+            let coin = vector::remove(&mut coins, 0);
```

Comment:
```suggestion
            let coin = vector::pop_back(&mut coins);
```

will be slightly more efficient--I don't think there's a reason we need to pull from the front.

Reply:
Yes, the order of the objectids doesn't matter for the call